### ISSUE TITLE
Panic: send on closed channel

### ISSUE THREAD

**Issue by sonyafenge** [CentaurusInfra/arktos#645]
**What happened**:
when run perf-tests with 5K+ nodes,  this panic happened in perf-tests output: nohup.out and also kubelet logs
```
goroutine 957017 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/watch.(*AggregatedWatcher).AddWatchInterface.func1(0x1a99600, 0xc056b75880, 0xc06afed580, 0xc056b75900)
        /home/sonyali/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/watch/watch.go:149 +0x1e9
created by k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/watch.(*AggregatedWatcher).AddWatchInterface
        /home/sonyali/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/watch/watch.go:134 +0xeb
```
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Arktos version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by sonyafenge** [comment]
https://github.com/futurewei-cloud/arktos/issues/572
https://github.com/futurewei-cloud/arktos/issues/571

**Comment by sonyafenge** [comment]
https://github.com/futurewei-cloud/arktos/pull/616

**Comment by panda98004** [comment]
A fix is working in uni-test

**Comment by jshaofuturewei** [comment]
In the article https://go101.org/article/channel-closing.html, it mentions channel closing principles.

A middle channel presenting all the data channel might be a solution.

**Comment by sonyafenge** [comment]
### 9/14/2020 [perf-tests][arktos][node5000] pr709-3a1w1e
![image](https://user-images.githubusercontent.com/45471193/93160883-2a307e00-f6c6-11ea-9f42-335c4e964585.png)

Logs can be can be found under GCP project: workload-controller-manager: sonya-useast1: /home/sonyali/logs/perf-test/gce-5000/arktos/0914-pr709-3a1w1e

**Comment by sonyafenge** [comment]
### 9/15/2020 [perf-tests][arktos][node5000] baseline-1a0w1e
5k runs yesterday to verify different issue, all of them using community ETCD3.4.4 with workload-controller disabled:
1.	baseline-1a0w1e with PR: Aggregated watcher recover #709 and PR: Initialize WorkloadInfo during defaulting (#706) 

> a.	Started from load testing, cluster crashed with “killing connection/stream” and panic in item b
> b.	panic in etcd.log: etcdserver/api/v3rpc/watch.go panic: runtime error: invalid memory address or nil pointer dereference #737
> c.	no repro for send on closed channel #645 and  Panic: "invalid memory address or nil pointer dereference" in daemon_controller.go #698 
2.	logs can be found under GCP project: workload-controller-manager on sonya-useast1: /home/sonyali/logs/perf-test/gce-5000/arktos/0915-baseline-1a0w1e

**Comment by sonyafenge** [comment]
### 9/15/2020 [perf-tests][arktos][node5000] baseline-1a0w1e
5k runs yesterday to verify different issue, all of them using community ETCD3.4.4 with workload-controller disabled:
1.	baseline-1a0w1e with PR: Aggregated watcher recover #709 and PR: Initialize WorkloadInfo during defaulting (#706) 

> a.	Started from load testing, cluster crashed with “killing connection/stream”
> b.	no repro for send on closed channel #645 and  Panic: "invalid memory address or nil pointer dereference" in daemon_controller.go #698

 
2.	logs can be found under GCP project: workload-controller-manager on sonya-useast1: /home/sonyali/logs/perf-test/gce-5000/arktos/0915-baseline-1a0w1e

**Comment by Sindica** [comment]
Resolving as this is no longer an issue.
